services/web/server/src/simcore_service_webserver/studies_dispatcher/_rest_handlers.py

```diff
@@ -18,12 +17,37 @@
 from ._catalog import ServiceMetaData, iter_latest_product_services
 from ._core import list_viewers_info
 from ._models import ViewerInfo
-from .handlers_redirects import (
-    compose_dispatcher_prefix_url,
-    compose_service_dispatcher_prefix_url,
-)
+from ._redirects_handlers import ViewerQueryParams
 
-logger = logging.getLogger(__name__)
+_logger = logging.getLogger(__name__)
+
+
+#
+# HELPERS to compose redirects
+#
+
+
+def _compose_file_and_service_dispatcher_prefix_url(
+    request: web.Request, viewer: ViewerInfo
+) -> HttpUrl:
+    """This is denoted PREFIX URL because it needs to append extra query parameters"""
+    params = ViewerQueryParams.from_viewer(viewer).dict()
+    absolute_url = request.url.join(
+        request.app.router["get_redirection_to_viewer"].url_for().with_query(**params)
+    )
+    return cast(HttpUrl, f"{absolute_url}")
```

Comment:
pare_obj_as would have done the trick nicelay as well and also validated the url, which might be waht you want here

Reply:
You are returning a string in place of an HttpUrl object. That's why mypy is complaining. Maybe at this point do it via HttpUrl the composition and validation or return a string.